background/main.ts

```diff
@@ -1668,8 +1668,36 @@ export default class Main extends BaseService<never> {
     return this.keyringService.exportPrivateKey(address)
   }
 
-  async importSigner(signerRaw: SignerRawWithType): Promise<HexString | null> {
-    return this.keyringService.importSigner(signerRaw)
+  async importSigner(
+    signerRaw: SignerRawWithType
+  ): Promise<{ success: boolean; errorMessage?: string }> {
```

Comment:
How about we make this an alternate?

```suggestion
  ): Promise<{ address: true } | { success: false; errorMessage: string }> {
```

Basically communicate "you will only see an error message in case of failure".

Reply:
I wanted to make this type this way but I have a problem making typescript happy about returning something more complicated than a simple object here 🤔 Something weird is going on with types for `createBackgroundAsyncThunk`, I will take a look later if I will have enough time.